### PR TITLE
(PUP-4865) Ensure error on yum install failure

### PR DESCRIPTION
**Pull request by MikaelSmith**
In older yum versions, attempting to install a package that doesn't
exist doesn't result in an error. Parse the yum output to identify
failure to install a package and throw an error.

The resulting error message is different on older vs newer ELs:
#### EL-7
```
Error: Execution of '/usr/bin/yum -d 0 -e 0 -y install not_a_package' returned 1: Error: Nothing to do
Error: /Stage[main]/Main/Package[not_a_package]/ensure: change from purged to present failed: Execution of '/usr/bin/yum -d 0 -e 0 -y install not_a_package' returned 1: Error: Nothing to do`
```

#### EL-5
```
Error: Could not find package not_a_package
Error: /Stage[main]/Main/Package[not_a_package]/ensure: change from purged to present failed: Could not find package not_a_package
```

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Verified acceptance test on centos 5 and centos 7 (x86_64).

File: acceptance/tests/resource/package/yum.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
+test_name "test the yum package provider"
+
+hosts_to_test = agents.select do |agent|
```
Comment (branan) [inline acceptance/tests/resource/package/yum.rb]: Should this just be a confine?

Comment (MikaelSmith) [inline acceptance/tests/resource/package/yum.rb]: Yes.

**Comment by MikaelSmith** [comment]
Updated with better confines.

**Comment by branan** [comment]
:+1: this looks reasonable to me

File: acceptance/tests/resource/package/yum.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  assert_match('Package[guid]/ensure: created', "#{result.host}: #{result.stdout}")
+end
+
+step 'Removing a specific version of a known package succeeds'
+verify_present agents, 'guid'
+apply_manifest_on(agents, 'package {"guid": ensure => absent}').each do |result|
+  assert_match('Package[guid]/ensure: removed', "#{result.host}: #{result.stdout}")
+end
+
+step 'Installing a non-existant version of a known package fails'
```
Comment (whopper) [inline acceptance/tests/resource/package/yum.rb]: Super nitpick that probably shouldn't hold up a merge: should be 'existent' :)

**Comment by whopper** [comment]
:+1: looks good to me as well

File: acceptance/tests/resource/package/yum.rb (hunk 2 above)
Comment (branan) [inline acceptance/tests/resource/package/yum.rb]: I don't care enough about this. I'm gonna push the button :stuck_out_tongue:

**Pull request merged**